staging/src/k8s.io/client-go/kubernetes/typed/certificates/v1beta1/certificatesigningrequest_expansion.go

```diff
@@ -31,6 +31,7 @@ type CertificateSigningRequestExpansion interface {
 func (c *certificateSigningRequests) UpdateApproval(ctx context.Context, certificateSigningRequest *certificates.CertificateSigningRequest, opts metav1.UpdateOptions) (result *certificates.CertificateSigningRequest, err error) {
 	result = &certificates.CertificateSigningRequest{}
 	err = c.client.Put().
+		GroupVersion(certificates.SchemeGroupVersion).
```

Comment:
maybe instead something like `VersionedAPIPath("/apis/certificates.k8s.io/v1beta1")` to keep Request limited to path prefixes and avoid repeated re-stringification of group versions?

Reply:
and calling that would recompute Request#pathPrefix (which currently is only done inside NewRequest?

Reply:
actually, to keep the AppsV1().RESTClient() method working as-is, I think we have to preserve the restclient held by each client knowledgable about APIPath and GroupVersion.

The real thing we want to avoid is constructing multiple transports. If we can create rest client based on another restclient that swaps out the groupversion/apipath, that could work